lua/nvim-treesitter-playground/query_linter.lua

```diff
@@ -0,0 +1,141 @@
+local api = vim.api
+local queries = require "nvim-treesitter.query"
+local ts_utils = require "nvim-treesitter.ts_utils"
+local utils = require "nvim-treesitter.utils"
+local configs = require "nvim-treesitter.configs"
+
+local hl_namespace = api.nvim_create_namespace("nvim-playground-lints")
+local ERROR_HL = "TSQueryLinterError"
+local MAGIC_NODE_NAMES = {"_", "ERROR"}
+
+local M = {}
+
+M.lints = {}
+M.use_virtual_text = true
+M.lint_events = {"BufWrite", "CursorHold"}
+
+local function lint_node(node, buf, error_type, complete_message)
+  if error_type ~= "Invalid Query" then
+    ts_utils.highlight_node(node, buf, hl_namespace, ERROR_HL)
+  end
+  local node_text = table.concat(ts_utils.get_node_text(node, buf), " ")
+  local error_text = complete_message or error_type .. ": " .. node_text
+  local error_range = {node:range()}
+  if M.use_virtual_text then
+    api.nvim_buf_set_virtual_text(buf, hl_namespace, error_range[1], {{error_text, ERROR_HL}}, {})
+  end
+  table.insert(M.lints[buf], {type = error_type, range = error_range, message = error_text, node_text = node_text})
+end
+
+local function table_contains(predicate, table)
+  for _, elt in pairs(table) do
+    if predicate(elt) then
+      return true
+    end
+  end
+  return false
+end
+
+function M.lint(buf)
+  buf = buf or api.nvim_get_current_buf()
+  M.clear_virtual_text(buf)
+  M.lints[buf] = {}
+
+  local filename = api.nvim_buf_get_name(buf)
+  local ok, query_lang = pcall(vim.fn.fnamemodify, filename, ":p:h:t")
+  query_lang = filename ~= '' and query_lang
```

Comment:
@steelsojka  Any ideas of how I should most elegantly get the language when I'm in the query buffer of the playground?